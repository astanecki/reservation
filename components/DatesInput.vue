<template>
  <div class="dates-input">
    <div class="dates-input__label">
      {{ labels.dates }}
    </div>

    <div class="dates-input__wrapper">
      <input
        :value="labels.checkIn"
        :class="{ 'dates-input--focused': focusedInputType === INPUT_TYPES.CHECK_IN }"
        @click="$emit('click', INPUT_TYPES.CHECK_IN)"
        @blur="$emit('blur')"
        readonly
      />
      <span class="dates-input__arrow">&#8594;</span>
      <input
        :value="labels.checkOut"
        :class="{ 'dates-input--focused': focusedInputType === INPUT_TYPES.CHECK_OUT }"
        @click="$emit('click', INPUT_TYPES.CHECK_OUT)"
        @blur="$emit('blur')"
        readonly
      />
    </div>
  </div>
</template>
<script>
/**
 * @enum {String}
 */
export const INPUT_TYPES = {
  CHECK_IN: 'check-in',
  CHECK_OUT: 'check-out',
};

export default {
  name: 'DatesInput',

  props: {
    focusedInputType: {
      type: String,
      default: '',
    }
  },

  data() {
    return {
      labels: {
        dates: 'Dates',
        checkIn: 'Check In',
        checkOut: 'Check Out',
      },
      INPUT_TYPES,
    }
  },
}
</script>
<style lang="scss" scoped>
.dates-input {
  &__label {
    text-align: left;
    font-weight: 700;
    color: #848484;
    font-size: 13px;
    margin-bottom: 5px;
  }

  &__wrapper {
    padding: 4px;
    display: flex;
    border: 1px solid #c6c6c6;
    margin-bottom: 15px;
  }

  &__arrow {
    font-size: 25px;
    margin-bottom: 7px;
    line-height: 15px;
    margin-right: 20px;
  }

  input {
    cursor: pointer;
    font-size: 15px;
    width: 50%;

    line-height: 15px;
    background-color: transparent;
    border: 0;

    &:focus {
      outline: none;
    }
  }

  & &--focused {
    outline: none;
    background-color: #8be7ce;
  }
}
</style>
