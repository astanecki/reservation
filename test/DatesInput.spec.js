import { mount } from "@vue/test-utils";

import DatesInput, { INPUT_TYPES } from "~/components/DatesInput";

describe("DatesInput", () => {
  const createWrapper = (propsData = {}) => {
    return mount(DatesInput, {
      propsData,
    });
  };

  it("is a Vue instance", () => {
    const wrapper = createWrapper();

    expect(wrapper.vm).toBeTruthy();
  });

  describe("named exports", () => {
    it(`should correctly export INPUT_TYPES`, () => {
      expect(INPUT_TYPES).toMatchInlineSnapshot(`
        Object {
          "CHECK_IN": "check-in",
          "CHECK_OUT": "check-out",
        }
      `);
    });
  });

  describe("snapshots", () => {
    it(`should match initial snapshot`, () => {
      const wrapper = createWrapper();

      expect(wrapper.vm.$el).toMatchSnapshot();
    });

    it(`should match snapshot when given prop is equal to INPUT_TYPES.CHECK_OUT`, () => {
      const wrapper = createWrapper({ focusedInputType: INPUT_TYPES.CHECK_OUT });

      expect(wrapper.vm.$el).toMatchSnapshot();
    });

    it(`should match snapshot when given prop is equal to INPUT_TYPES.CHECK_IN`, () => {
      const wrapper = createWrapper({ focusedInputType: INPUT_TYPES.CHECK_IN });

      expect(wrapper.vm.$el).toMatchSnapshot();
    });
  });

  describe("data", () => {
    it(`should prepare basic data correctly`, () => {
      expect(DatesInput.data()).toMatchInlineSnapshot(`
        Object {
          "INPUT_TYPES": Object {
            "CHECK_IN": "check-in",
            "CHECK_OUT": "check-out",
          },
          "labels": Object {
            "checkIn": "Check In",
            "checkOut": "Check Out",
            "dates": "Dates",
          },
        }
      `);
    });
  });

  describe("events", () => {
    describe("click", () => {
      let wrapper;

      beforeEach(() => {
        wrapper = createWrapper();

        wrapper.vm.$emit = jest.fn();
      });

      it(`should be called when click event fired on first input`, () => {
        wrapper.findAll('input').at(0).trigger('click');

        expect(wrapper.vm.$emit).toHaveBeenCalledWith('click', INPUT_TYPES.CHECK_IN);
      });

      it(`should be called when click event fired on second input`, () => {
        wrapper.findAll('input').at(1).trigger('click');

        expect(wrapper.vm.$emit).toHaveBeenCalledWith('click', INPUT_TYPES.CHECK_OUT)
      });
    });
  });
});
