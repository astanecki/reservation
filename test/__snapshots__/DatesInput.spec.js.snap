// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DatesInput snapshots should match initial snapshot 1`] = `
<div
  class="dates-input"
>
  <div
    class="dates-input__label"
  >
    
    Dates
  
  </div>
   
  <div
    class="dates-input__wrapper"
  >
    <input
      class=""
      readonly="readonly"
    />
     
    <span
      class="dates-input__arrow"
    >
      →
    </span>
     
    <input
      class=""
      readonly="readonly"
    />
  </div>
</div>
`;

exports[`DatesInput snapshots should match snapshot when given prop is equal to INPUT_TYPES.CHECK_IN 1`] = `
<div
  class="dates-input"
>
  <div
    class="dates-input__label"
  >
    
    Dates
  
  </div>
   
  <div
    class="dates-input__wrapper"
  >
    <input
      class="dates-input--focused"
      readonly="readonly"
    />
     
    <span
      class="dates-input__arrow"
    >
      →
    </span>
     
    <input
      class=""
      readonly="readonly"
    />
  </div>
</div>
`;

exports[`DatesInput snapshots should match snapshot when given prop is equal to INPUT_TYPES.CHECK_OUT 1`] = `
<div
  class="dates-input"
>
  <div
    class="dates-input__label"
  >
    
    Dates
  
  </div>
   
  <div
    class="dates-input__wrapper"
  >
    <input
      class=""
      readonly="readonly"
    />
     
    <span
      class="dates-input__arrow"
    >
      →
    </span>
     
    <input
      class="dates-input--focused"
      readonly="readonly"
    />
  </div>
</div>
`;
